<a href="https://colab.research.google.com/github/codemishka/Prediction-of-Product-Sales/blob/main/1689878241343_Untitled6__1_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Loading Data

# Imports

In [8]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [9]:

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Loading Data

In [12]:
fname='/content/drive/MyDrive/CodingDojo/01-Fundamentals/Week02/Data/sales_predictions_2023.csv'

In [13]:
# load in the data as a pandas dataframe
df = pd.read_csv(fname)

In [14]:
df.info() and df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


# Data Cleaning

The data has 8523 rows and 12 Columns

In [15]:
df.shape

(8523, 12)

In [16]:
df.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

There are no unnamed columns.

All  data types are as they should be.

# Duplicated Data

In [17]:
df.drop_duplicates(inplace = True)
df.duplicated().sum()

0

There are no duplicated rows.

# Missing Values

Inspecting Categorical Columns & Addressing Inconsistent Values

In [18]:
df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [ ]:
#Strategy #1: To deal with the missing values, we could simply drop the missing values.

In [19]:
df.dropna(subset=['Item_Weight'], inplace=True)

In [20]:
df.dropna(subset=['Outlet_Size'], inplace=True)

checking the missing values

In [21]:

df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

# Find and fix any inconsistent categories of data (for example: fix cat, Cat, and cats so that they are consistent).

#TBC

# Summary statistics of each (min, max, mean).

In [22]:
#Summary Statistics for Numeric Features
stats =  df.describe()
stats.loc[['mean','min','max']]


Item_Weight  Item_Visibility    Item_MRP  Outlet_Establishment_Year  \
mean    12.898675         0.060700  141.716328                1999.190538   
min      4.555000         0.000000   31.490000                1987.000000   
max     21.350000         0.188323  266.888400                2009.000000   

      Item_Outlet_Sales  
mean        2272.037489  
min           69.243200  
max        10256.649000